app.rb

```diff
@@ -0,0 +1,174 @@
+      create_teacher
+    else
+      colorize_output(31, 'Invalid choice')
+    end
+  end
+
+  def create_student
+    colorize_outprint(35, 'Name: ')
+    name = gets.chomp
+    colorize_outprint(35, 'Age: ')
+    age = gets.chomp
+    colorize_outprint(35, 'Has parent permission? [Y/N]: ')
+    permission = gets.chomp.downcase
+    case permission
+    when 'y' then parent_permission = true
+    when 'n' then parent_permission = false
+    end
+    student = Student.new(age: age, name: name, parent_permission: parent_permission)
+    @people.push(student)
+    colorize_output(34, 'Person Student Created successfully')
+    display_list
+    continue
+  end
+
+  def create_teacher
+    colorize_outprint(35, 'Name: ')
+    name = gets.chomp
+    colorize_outprint(35, 'Age: ')
+    age = gets.chomp
+    colorize_outprint(35, 'Specialization: ')
+    specialization = gets.chomp
+    @people.push(Teacher.new(age: age, name: name, specialization: specialization))
+    colorize_output(34, 'Person Teacher Created successfully')
+    display_list
+    continue
+  end
+
+  def create_book
+    colorize_output(36, 'Enter Book details')
+    colorize_outprint(35, 'Title: ')
+    title = gets.chomp
+
+    colorize_outprint(35, 'Author: ')
+    author = gets.chomp
+
+    @books.push(Book.new(title, author))
+
+    colorize_output(36, 'You have successfully created your book')
+    display_list
+    continue
+  end
+
+  def list_all_books_with_numbers
+    if @books.empty?
+      colorize_output(34, 'Please enter books first!')
+      display_list
+      continue
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_all_person_with_numbers
+    if @people.empty?
+      colorize_output(34, 'Please enter people first!!')
+      display_list
+      continue
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_rental()
+    colorize_output(36, 'Select a book from the following list by number')
+    list_all_books_with_numbers
+    book_option = gets.chomp.to_i
+
+    colorize_output(36, 'Select a person from the following list by number (not id)')
+    list_all_person_with_numbers
+    person_option = gets.chomp.to_i
+
+    colorize_outprint(36, 'Date: ')
+    date = gets.chomp
+    @rentals.push(Rental.new(date, @books[book_option], @people[person_option]))
+    colorize_output(36, 'Rental created successfully')
+    display_list
+    continue
+  end
+
+  def list_rentals_of_person_id()
+    print 'Person ID: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'List of Rentals:'
+    rentals.each do |rental|
+      puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+    end
+    display_list
+    continue
+  end
```

Comment:
- [ ] When I try to display a rental for a person, the app crashes with the error in the screenshot below. Kindly fix that 👍🏽

<img width="1353" alt="Screenshot 2023-03-17 at 12 06 22 PM" src="https://user-images.githubusercontent.com/50941074/225900159-52d8c652-c634-4335-bc0d-e1c8abdda12e.png">